### DISCUSSION TITLE
Quotes in output if string is number

### DISCUSSION THREAD

**Discussion by paradajz** [TomWright/dasel#91] [Q&A] [answered]
Given the following YAML example:

```
  analog:
    type: "native"
```

Running `dasel -n -f demo.yml analog.type` gives me `native`. No quotes. However, if type is something like this:

```
  analog:
    type: "4067"
```

Running the same command gives me `"4067"`. Notice the quotes in output. Is this a bug or am I missing something here?

**Comment by TomWright** [comment, marked as answer]
Hey @paradajz - This isn't a bug.

Since you are referencing a YAML file the input + output is being parsed/formatted as YAML.

- `"native"` is a string
- `native` is a string
- `"4067"` is a string
- `4067` is an *int*

In order for dasel to output 4067 as a string it must be surrounded by quotes.

If you do not want the output to be formatted as yaml you can use the  `-w -` flag or `--plain` flag.

These are equivalent for this use-case and will cause the output to be `native` and `4067` without quotes.

``` bash
$ dasel -n -f demo.yml --plain analog.type
4067
```
